<a href="https://colab.research.google.com/github/Pradip240/IPL-T20-Cricket-Analysis/blob/main/Ameen_IPL_T20_Cricket_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Indian Premier League(IPL) is a professional Twenty20 cricket league in India contested during March or April and May of every year by eight teams representing eight different cities in India. The league was founded by the Board of Control for Cricket in India(BCCI) in 2008. </b>

## <b> Explore and analyze the data to discover results and statistics for different teams playing in IPL. </b>

##Let's load and explore the dataset

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
#Importing the libraries.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [ ]:
file_path = '/content/drive/MyDrive/AlmaBetter Data Science/1. PYTHON/CAPSTONE PROJECT/IPL Data/'
ipl_df = pd.read_csv(file_path + 'matches.csv')
ipl_df1 = pd.read_excel(file_path + 'Players.xlsx')
ipl_df2 = pd.read_csv(file_path + 'deliveries.csv')
ipl_df3 = pd.read_csv(file_path + 'teams.csv')
ipl_df4 = pd.read_csv(file_path + 'most_runs_average_strikerate.csv')
ipl_df5 = pd.read_csv(file_path + 'teamwise_home_and_away.csv')

In [ ]:
#Checking the shape multiple datasets
df_ipl=[ipl_df.shape, ipl_df1.shape, ipl_df2.shape, ipl_df3.shape, ipl_df4.shape, ipl_df5.shape]
df_ipl


[(756, 18), (566, 5), (179078, 21), (15, 1), (516, 6), (14, 7)]

# So what can we expect from the multiple IPL datasets?
##The datasets are:
###1.matches.csv
###2.Players.xlsx
###3.deliveries.csv
###4.teams.csv
###5.most_runs_average_strikerate.csv
###6.teamwise_home_and_away.csv

###In order to understand our data, we can look at each variable and try to understand their meaning and relevance to this problem.
###The datsets consists of multiple observations & columns. Let's go ahead & get familiar with the columns.

#Cleaning the dataset

##From matches dataset we have:-
###**ID:** Basically the IPL match ID.
###**Season:** The IPL season, which in our dataset if from season 2008 - season 2019
###**City:**The cities where the IPL matches were played.
###**Date:** The date on which the matches were played or held.
###**Team1 & Team2:** The teams participated in the IPL matches since 2008.
###**Toss_winner:** The team that won the toss.
###**Toss_decision:** The decision taken by the toss winning team either to 'Bat' or 'Field' first.
###**Result:** The result column consists of (normal, tie, no result) of the match.
###**Dl_applied:** The column consists the numerical values (0 or 1) which indicates whether the Duckworth-Lewis method was applied or not.
###**Winner:** Winner of the respective match.
###**Win_by_runs:** The runs by which the team batting first won.
###**Win_by_wickets:** The wickets by which the team batting second won.
###**Player_of_the_match:** The player who performed exceptionally good in the match.
###**Venue:** The venue where the match was hosted.
###**Umpire1 & Umpire2:** Two on-field umpires during the match.
###**Umpire3:** Basically the third Umpire.

#Matches Dataset

In [ ]:
#Checking the first five observations of matches dataset
matches=ipl_df.head()
matches

In [ ]:
ipl_df.info()

In [ ]:
#Checking for the null values in the data
missing_val = ipl_df.isnull()
missing_val

In [ ]:
#suming up the null values together
ipl_df.isnull().sum()

In [ ]:
#Plotting the heatmap for the missing values in the matches dataset
plt.figure(figsize=(20,15))
sns.heatmap(missing_val)

In [ ]:
#Filling up the numerical missing values
filling_na = ipl_df.fillna(ipl_df.mean(axis=0))

In [ ]:
filled_val = filling_na.isnull()

In [ ]:
#Plotting the heatmap after filling up the missing numerical values
plt.figure(figsize=(10,10))
sns.heatmap(filled_val)

In [ ]:
#Filling up the categorical missing values
filling_na_cat = ipl_df.fillna(ipl_df.mode().iloc[0])

In [ ]:
filling_na_cat.isnull().sum()

In [ ]:
filled_val1 = filling_na_cat.isnull()

In [ ]:
#Plotting the heatmap after filling up the missing categorical values
plt.figure(figsize=(10,10))
sns.heatmap(filled_val1)

#Players Dataset

##From the players dataset:-(rows 556, columns 5)
###*It consists of following columns- player_name, DOB, Batting_skill, Bowling_skill & Country

In [ ]:
#Checking the first five observations of players dataset
players = ipl_df1.head()
players

In [ ]:
ipl_df1.info()

In [ ]:
#Suming up all the missing values
ipl_df1.isnull().sum()

#As we can see all the missing values are categorical

In [ ]:
show = ipl_df1.isnull()

In [ ]:
#Plotting the heatmap for the missing values in the players dataset
plt.figure(figsize=(10,10))
sns.heatmap(show)


In [ ]:
#Filling up the caetgorical missing values
filling_na_cat1 = ipl_df1.fillna(ipl_df1.mode().iloc[0])

In [ ]:
show1 = filling_na_cat1.isnull()

In [ ]:
#Plotting the heatmap after filling up the missing categorical values
plt.figure(figsize=(10,10))
sns.heatmap(show1)

#Deliveries Dataset

##From the deliveries dataset:-(rows 179078, columns 21)
###*It consists of following columns- match_id, inning, batting_team,	bowling_team,	over,	ball,	batsman,	non_striker,	bowler,	is_super_over,	wide_runs,	bye_runs,	legbye_runs,	noball_runs,	penalty_runs,	batsman_runs,	extra_runs,	total_runs,	player_dismissed,	dismissal_kind,	fielder.



In [ ]:
#Checking the first five observations of deliveries dataset
ipl_df2.head()

In [ ]:
ipl_df2.info()

In [ ]:
#Summing up the null values
ipl_df2.isnull().sum()

In [ ]:
show2 = ipl_df2.isnull()

In [ ]:
#Plotting the heatmap for the missing values in the deliveries dataset
plt.figure(figsize=(10,10))
sns.heatmap(show2)

##As seen above the last three columns has significant amount of missing values, so here we can drop those three columns.

In [ ]:
#Dropping the last three columns
ipl_df2 = ipl_df2.drop(columns =['player_dismissed', 'dismissal_kind', 'fielder'], axis=1)

In [ ]:
ipl_df2.head()

In [ ]:
show3 = ipl_df2.isnull()

In [ ]:
#Plotting the heatmap after dropping off the missing categorical values
plt.figure(figsize=(10,10))
sns.heatmap(show3)

#Teams Dataset

##From teams dataset:- (rows 15, column 1)
###*Just has the team names in it.

In [ ]:
#Checking the first ten observations of teams dataset
ipl_df3.head(10)

In [ ]:
ipl_df3.info()

In [ ]:
ipl_df3.isnull().sum()

team1    0
dtype: int64

##Teams dataset has 'zero' null values.

#Most_runs_average_strikerate Dataset

---




##From most_runs_average_strikerate dataset:- (rows 516, columns 6)
###*It consists of following columns- batsman, total_runs, out,	numberofballs, average, strikerate.

In [ ]:
#Checking the first ten observations of most_runs_average_strikerate dataset
ipl_df4.head(10)

In [ ]:
ipl_df4.info()

In [ ]:
ipl_df4.isnull().sum()

In [ ]:
show4 = ipl_df4.isnull()

In [ ]:
#Plotting the heatmap for the missing values in the dataset
plt.figure(figsize=(10,5))
sns.heatmap(show4)

In [ ]:
#Treating the missing numerical values
filling_num = ipl_df4.fillna(ipl_df4.mean(axis=0))

In [ ]:
show5 = filling_num.isnull()

In [ ]:
#Plotting the heatmap for the missing numerical values in the dataset
plt.figure(figsize=(10,5))
sns.heatmap(show5)

#Teamwise_home_and_away Dataset

##From teamwise_home_and_away datset:- (rows 14, columns 7)
###*It consists of following columns- team, home_wins, away_wins, home_matches	away_matches,	home_win_percentage,	away_win_percentage.

In [ ]:
#Checking the first ten observations of teamwise_home_and_away dataset
ipl_df5.head(10)


In [ ]:
ipl_df5.info()

In [ ]:
ipl_df5.isnull().sum()

##The dataset above has no missing values to treat

##As our dataset has been cleaned, we can now do the in-depth analysis and visualization.